 # Table of Contents
<div class="toc" style="margin-top: 1em;"><ul class="toc-item" id="toc-level0"><li><span><a href="http://localhost:8888/notebooks/07/ukoly/07_reseni.ipynb#Plnící-stroj" data-toc-modified-id="Plnící-stroj-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Plnící stroj</a></span></li></ul></div>

# Úkoly

K vypracování těchto úkolů budete potřebovat některé základní znalosti z předchozích lekcí, ale především ty z této lekce.

## Plnící stroj

Ve vašem rodinném pivovaru máte prastarý plnící stroj lahví a chcete zjistit, jak přesně tento starý vrak vlastně lahve plní. Za tímto účelem jste po naplnění 20 lahví změřili objem jejich obsahu. Stanovte na základě vašich měření očekávaný rozsah objemů vašich piv s pravděpodobností 99%. Vaše data najdete v souboru [`plnici_stroj.csv`](plnici_stroj.csv). Údaje jsou v ml.

In [1]:
mereni <- read.table("plnici_stroj.csv")
head(mereni)

Volume
1 484.11
2 459.49
3 471.38
4 512.01
5 494.48
6 528.63

In [2]:
mereni <- mereni$Volume
class(mereni)
length(mereni)

[1] "numeric"

[1] 20

Manuálně:

In [3]:
sem <- sd(mereni) / sqrt(length(mereni))
t <- qt(0.995, df = length(mereni) - 1)
mean(mereni) - t * sem
mean(mereni) + t * sem

[1] 475.7094

[1] 507.4316

pomocí funkce `t.test`

In [4]:
t.test(mereni, conf.level = 0.99)


	One Sample t-test

data:  mereni
t = 88.667, df = 19, p-value < 2.2e-16
alternative hypothesis: true mean is not equal to 0
99 percent confidence interval:
 475.7094 507.4316
sample estimates:
mean of x 
 491.5705 


Náš starý stroj tedy naplní 99% procent lahví někde mezi 475.7 ml a 507.4. 

Pokud by nás zajímala pouze spodní hranice, tedy interval od minimálního možného naplnění do nekonečna, můžeme tak učinit takto:

In [5]:
t <- qt(0.99, df = length(mereni) - 1)
mean(mereni) - t * sem

[1] 477.4915

nebo opět funkci `t.test` ale s parametrem `alternative`, který specifikuje o jakou alternativu se jedná (více v další lekci):

In [6]:
t.test(mereni, conf.level = 0.99, alternative = "greater")


	One Sample t-test

data:  mereni
t = 88.667, df = 19, p-value < 2.2e-16
alternative hypothesis: true mean is greater than 0
99 percent confidence interval:
 477.4915      Inf
sample estimates:
mean of x 
 491.5705 


Náš stroj tedy v 99% případů zákazníky ošidí maximálně asi o 22.5 ml lahodného moku.